cli_tools/common/image/importer/inflater.go

```diff
@@ -84,46 +105,79 @@ type inflaterFacade struct {
 	apiInflater   Inflater
 	daisyInflater Inflater
 	logger        logging.Logger
+	qemuChecksum  string
+	computeClient daisyCompute.Client
+	request       ImageImportRequest
 }
 
-func (facade *inflaterFacade) Inflate() (persistentDisk, shadowTestFields, error) {
+func (facade *inflaterFacade) Inflate() (persistentDisk, inflationInfo, error) {
 	var pd persistentDisk
-	var tf shadowTestFields
+	var ii inflationInfo
 	var err error
+	var fallbackReason string
 
-	//  Try with API inflater.
-	pd, tf, err = facade.apiInflater.Inflate()
-	if err == nil {
-		facade.logger.Metric(&pb.OutputInfo{
-			InflationType:   "api_success",
-			InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-		})
-		return pd, tf, err
-	}
+	if facade.qemuChecksum == "" {
+		fallbackReason = "qemu_checksum_missing"
+	} else {
+		//  Run API inflater as the primary inflation method.
+		pd, ii, err = facade.apiInflater.Inflate()
 
-	if !isCausedByUnsupportedFormat(err) {
-		facade.logger.Metric(&pb.OutputInfo{
-			InflationType:   "api_failed",
-			InflationTimeMs: []int64{tf.inflationTime.Milliseconds()},
-		})
-		return pd, tf, err
+		if err != nil {
+			// If the inflation is failed but not due to unsupported format, don't rerun inflation.
+			if !isCausedByUnsupportedFormat(err) {
+				facade.logger.Metric(&pb.OutputInfo{
+					InflationType:   "api_failed",
+					InflationTimeMs: []int64{ii.inflationTime.Milliseconds()},
+				})
+				return pd, ii, err
+			}
+			fallbackReason = "unsupported_format"
+		}
+
+		if err == nil {
+			if isChecksumMatch(ii.checksum, facade.qemuChecksum) {
+				facade.logger.Metric(&pb.OutputInfo{
+					InflationType:   "api_success",
+					InflationTimeMs: []int64{ii.inflationTime.Milliseconds()},
+				})
+				return pd, ii, err
+			}
+
+			// If checksum mismatches , delete the corrupted disk.
+			facade.logger.User("Disk checksum mismatch, recreating...")
```

Comment:
Why does this log state "recreating" when it's deleting a disk? Does that part of the log message apply to the code outside this block (`fallbackToDaisyInflater')? If so, move it after the deleteDisk attempt, since if disk deletion fails, then this log does not apply.

Reply:
you are right, moved